sdk/Trace/TraceContext.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use OpenTelemetry\Trace as API;
+
+final class TraceContext implements API\TextMapFormatPropagator
+{
+    public const TRACEPARENT = 'http_traceparent';
+    public const TRACESTATE = 'http_tracestate';
+
+    // TODO Consolidate these and the validity checks in SpanContext.php
+    private const SUPPORTED_VERSION = '00';
+    private const INVALID_TRACE = '00000000000000000000000000000000';
+    private const INVALID_SPAN = '0000000000000000';
+    private const VALID_VERSION = '/^[0-9a-f]{2}$/';
+    private const VALID_TRACE = '/^[0-9a-f]{32}$/';
+    private const VALID_SPAN = '/^[0-9a-f]{16}$/';
+    private const VALID_TRACEFLAGS = '/^[0-9a-f]{2}$/';
+    private const SAMPLED_FLAG = 1;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function fields(): array
+    {
+        return [self::TRACEPARENT, self::TRACESTATE];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function inject(API\SpanContext $context, &$carrier, API\PropagationSetter $setter): void
+    {
+        $traceparent = self::SUPPORTED_VERSION . '-' . $context->getTraceId() . '-' . $context->getSpanId() . '-' . ($context->isSampled() ? '01' : '00');
+        $setter->set($carrier, self::TRACEPARENT, $traceparent);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function extract($carrier, API\PropagationGetter $getter): API\SpanContext
+    {
+        $traceparent = $getter->get($carrier, self::TRACEPARENT);
+        if (null === $traceparent) {
+            throw new \InvalidArgumentException('Traceparent not present');
+        }
+
+        // Traceparent = {version}-{trace-id}-{parent-id}-{trace-flags}
+        $pieces = explode('-', $traceparent);
+
+        $peicesCount = count($pieces);
+        if ($peicesCount != 4) {
+            throw new \InvalidArgumentException('Unable to extract traceparent. Contains Invalid values');
```

Comment:
Maybe we could make this exception mention that there are an incorrect number of peices?